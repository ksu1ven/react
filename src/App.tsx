import SearchPage from './components/SearchPage/SearchPage';
import ErrorBoundary from './components/ErrorBoundary';

function App() {
  return (
    <ErrorBoundary>
      <SearchPage />
    </ErrorBoundary>
  );
}

export default App;
