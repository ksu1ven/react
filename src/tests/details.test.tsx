import { beforeAll, describe, expect, it } from 'vitest';
import '@testing-library/jest-dom';
import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import App from '../App';

describe('Tests for the Detailed Card component', () => {
  beforeAll(async () => {
    //Click card to get search params - ?details=testCard-0 -  for thesee three tests"
    render(<App />);
    await waitFor(() => {
      const card = screen.getAllByTestId('card')[0];
      fireEvent.click(card);
      screen.getByTestId('details');
      cleanup();
    });
  });

  it('Loading indicator is displayed while fetching data', async () => {
    expect(location.search).toBe('?details=testCard-0');
    render(<App />);
    expect(screen.getByTestId('details')).toContainHTML(
      '<span class="loader" data-testid="loader" />'
    );
  });

  it('Detailed card component correctly displays the detailed card data;', async () => {
    render(<App />);
    expect(location.search).toBe('?details=testCard-0');
    const details = await screen.findByTestId('details');
    expect(details).toMatchSnapshot();
    expect(screen.getByTestId('details')).toBeInTheDocument();
    expect(await screen.findByTestId('details-h1')).toHaveTextContent(
      'postRequestCardName'
    );
    expect(screen.getByText('EarthAnimal: No')).toBeInTheDocument();
    expect(screen.getByText('Avian: Yes')).toBeInTheDocument();
  });

  it('Clicking the close button hides the component.', async () => {
    render(<App />);
    expect(location.search).toBe('?details=testCard-0');
    expect(await screen.findByTestId('details')).toBeInTheDocument();
    fireEvent.click(screen.getByTestId('cross'));
    const details = await screen.findByTestId('details');
    expect(details).toMatchSnapshot();
    expect(screen.queryByTestId('details')).not.toBeInTheDocument();
    expect(screen.queryByTestId('details-h1')).not.toBeInTheDocument();
    expect(screen.queryByText('EarthAnimal: No')).not.toBeInTheDocument();
    expect(screen.queryByText('Avian: Yes')).not.toBeInTheDocument();
  });
});
