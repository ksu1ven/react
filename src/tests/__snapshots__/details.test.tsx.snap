// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the Detailed Card component > Clicking the close button hides the component. 1`] = `
<aside
  class="relative w-1/3 bg-lime-700 text-white flex flex-col pt-40 items-center gap-6"
  data-testid="details"
>
  <button
    class="absolute top-14 right-14 flex items-center justify-center w-16 h-16 bg-lime-300 text-4xl p-3 rounded-full"
    data-testid="cross"
  >
    X
  </button>
  <span
    class="loader"
    data-testid="loader"
  />
</aside>
`;

exports[`Tests for the Detailed Card component > Clicking the close button hides the component. 2`] = `null`;

exports[`Tests for the Detailed Card component > Detailed card component correctly displays the detailed card data; 1`] = `
<aside
  class="relative w-1/3 bg-lime-700 text-white flex flex-col pt-40 items-center gap-6"
  data-testid="details"
>
  <button
    class="absolute top-14 right-14 flex items-center justify-center w-16 h-16 bg-lime-300 text-4xl p-3 rounded-full"
    data-testid="cross"
  >
    X
  </button>
  <div>
    <h1
      class="text-5xl font-extrabold text-lime-300 mb-10"
      data-testid="details-h1"
    >
      testCard-0
    </h1>
    <ul
      class="text-3xl font-extrabold"
    >
      <li>
        Avian: 
        Yes
      </li>
      <li>
        Canine: 
        No
      </li>
      <li>
        EarthAnimal: 
        No
      </li>
      <li>
        EarthInsect: 
        No
      </li>
      <li>
        Feline: 
        No
      </li>
    </ul>
  </div>
</aside>
`;

exports[`Tests for the Detailed Card component > Loading indicator is displayed while fetching data 1`] = `
<aside
  class="relative w-1/3 bg-lime-700 text-white flex flex-col pt-40 items-center gap-6"
  data-testid="details"
>
  <button
    class="absolute top-14 right-14 flex items-center justify-center w-16 h-16 bg-lime-300 text-4xl p-3 rounded-full"
    data-testid="cross"
  >
    X
  </button>
  <span
    class="loader"
    data-testid="loader"
  />
</aside>
`;
